src/cpp/client/channel.h

```diff
@@ -56,12 +56,39 @@ class Channel GRPC_FINAL : public GrpcLibrary, public ChannelInterface {
   Channel(const grpc::string& host, grpc_channel* c_channel);
   ~Channel() GRPC_OVERRIDE;
 
-  virtual void* RegisterMethod(const char* method) GRPC_OVERRIDE;
-  virtual Call CreateCall(const RpcMethod& method, ClientContext* context,
+  void* RegisterMethod(const char* method) GRPC_OVERRIDE;
+  Call CreateCall(const RpcMethod& method, ClientContext* context,
                           CompletionQueue* cq) GRPC_OVERRIDE;
-  virtual void PerformOpsOnCall(CallOpSetInterface* ops,
+  void PerformOpsOnCall(CallOpSetInterface* ops,
                                 Call* call) GRPC_OVERRIDE;
 
+  grpc_connectivity_state GetState(bool try_to_connect) GRPC_OVERRIDE;
+
+  void NotifyOnStateChange(grpc_connectivity_state last_observed,
+                           gpr_timespec deadline,
+                           CompletionQueue* cq, void* tag) GRPC_OVERRIDE;
+
+  bool WaitForStateChange(grpc_connectivity_state last_observed,
+                          gpr_timespec deadline) GRPC_OVERRIDE;
+
+  bool WaitForState(grpc_connectivity_state target_state,
+                    gpr_timespec deadline) GRPC_OVERRIDE;
+
+#ifndef GRPC_CXX0X_NO_CHRONO
+  void NotifyOnStateChange(
+      grpc_connectivity_state last_observed,
+      const std::chrono::system_clock::time_point& deadline,
+      CompletionQueue* cq, void* tag) GRPC_OVERRIDE;
+
+  bool WaitForStateChange(
+      grpc_connectivity_state last_observed,
+      const std::chrono::system_clock::time_point& deadline) GRPC_OVERRIDE;
+
+  bool WaitForState(grpc_connectivity_state target_state,
```

Comment:
I would prefer a standalone WaitForState (that takes a Channel as an additional param) as a separate utility. This keeps our API orthogonal and still gives people the right convenience function. Separating that out in a separate header also avoids promoting its unnecessary usage.